### ISSUE TITLE
Mapping from java.util.Properties if key contains dot is not possible

### ISSUE THREAD

**Issue by typekpb** [mapstruct/mapstruct#3523]
### Use case

Having method like:

```
@Data
public class Bar {
  String foo;

}

@Mapper
public interface BarMapper {
  @Mapping( target="foo", source="x.y")
  Bar map(java.util.Properties source);
}
```

results in error:
```
 The type of parameter "source" has no property named "x.y"
```

### Possible workarounds

If source map key String contains dot "." it's misinterpreted as transient reference. I haven't found any workaround for the problem.

### MapStruct Version

1.5.5.Final

**Comment by filiphr** [comment]
Closing this as a duplicate of #3066

**Comment by filiphr** [comment]
Apart from the fact that we do not support keys in source property names (see #3066), we also do not support `Properties`, what we do support is `Map<String, ?>`, but `Properties` is `Map<Object, Object>` so even when we fix #3066 mapping from `Properties` will not work. 

If you want `Properties` to be supported will need a dedicated feature request for that
